server/scripts/webscraper.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python3
+
+import re
+import requests
+import urllib.request
+import time
+from bs4 import BeautifulSoup
+
+# Started by Jeremy Lim on 20/07/2019
+
+# TO-DO LIST
+# - Get timetable done
+# - Get GenEd done
+# - Majors
+# - Degrees
+# - Add to database
+
+# Stripped down alphabet
+course_alphabet = ['A','B','C','D','E','F','G','H','I','L','M','N','O','P','R','S','T','V','Y','Z']
+spec_alphabet = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','R','S','T','V','W']
+
+##### COURSES #####
+
+# To print the info of each course
+def print_course(code, link, name, cred):
+    print("CODE: " + code) # prints the code of the course
+    print("LINK: " + link) # prints the link of the course
+    print("NAME: " + name) # prints the name of the course
+    print("CREDITS: " + cred) # prints the credits of the course
+
+# To go through each letter's links for courses
+def run_course():
+
+    for letter in course_alphabet:
+        course_url = 'http://legacy.handbook.unsw.edu.au/vbook2018/brCoursesByAtoZ.jsp?StudyLevel=Undergraduate&descr=' + letter
+        response = requests.get(course_url)
+        course_soup = BeautifulSoup(response.text, "html.parser")
+
+        # Do webscraping
+        tr = course_soup.find_all('tr')
+        for i in range(1,3):
+            counter = 0
+            td = tr[i].find_all('td')
+            code = td[0].text
+            link = td[1].find_all('a')[0]['href']
+            name = td[1].find_all('a')[0].text
+            cred = td[2].text
+
+            print_course(code, link, name, cred)
+
+            # Go to course link and scrape the data
```

Comment:
this enters each course's link and scrapes all the relevant data, some stuff like term offerings are still not done because it requires some extra tests